sky/data/skystorage.py

```diff
@@ -0,0 +1,334 @@
+    _CURSOR.execute('SELECT csync_pid FROM running_csync WHERE source_path=(?)',
+                    (path,))
+    row = _CURSOR.fetchone()
+    if row:
+        return row[0]
+    raise ValueError(f'{path} is not mounted with CSYNC.')
+
+
+@click.group()
+def main():
+    pass
+
+
+def update_interval(interval: int, elapsed_time: int):
+    """Updates the time interval for the next sync operation.
+
+    Given the originally set interval and the time elapsed during the
+    sync operation, this function computes and returns the remaining time to
+    wait before the next sync operation.
+    """
+    diff = interval - elapsed_time
+    if diff <= 0:
+        return 0
+    else:
+        return diff
+
+
+def get_s3_upload_cmd(src_path: str, dst: str, num_threads: int, delete: bool,
+                      no_follow_symlinks: bool):
+    """Builds sync command for aws s3"""
+    config_cmd = ('aws configure set default.s3.max_concurrent_requests '
+                  f'{num_threads}')
+    subprocess.check_output(config_cmd, shell=True)
+    sync_cmd = f'aws s3 sync {src_path} s3://{dst}'
+    if delete:
+        sync_cmd += ' --delete'
+    if no_follow_symlinks:
+        sync_cmd += ' --no-follow-symlinks'
+    return sync_cmd
+
+
+def get_gcs_upload_cmd(src_path: str, dst: str, num_threads: int, delete: bool,
+                       no_follow_symlinks: bool):
+    """Builds sync command for gcp gcs"""
+    sync_cmd = (f'gsutil -m -o \'GSUtil:parallel_thread_count={num_threads}\' '
+                'rsync -r')
+    if delete:
+        sync_cmd += ' -d'
+    if no_follow_symlinks:
+        sync_cmd += ' -e'
+    sync_cmd += f' {src_path} gs://{dst}'
+    return sync_cmd
+
+
+def run_sync(src: str,
+             storetype: str,
+             dst: str,
+             num_threads: int,
+             interval: int,
+             delete: bool,
+             no_follow_symlinks: bool,
+             csync_pid: int,
+             max_retries: int = 10,
+             backoff: Optional[common_utils.Backoff] = None):
+    """Runs the sync command to from src to storetype bucket"""
+    #TODO: add enum type class to handle storetypes
+    storetype = storetype.lower()
+    if storetype == 's3':
+        sync_cmd = get_s3_upload_cmd(src, dst, num_threads, delete,
+                                     no_follow_symlinks)
+    elif storetype == 'gcs':
+        sync_cmd = get_gcs_upload_cmd(src, dst, num_threads, delete,
+                                      no_follow_symlinks)
+    else:
+        raise ValueError(f'Unsupported store type: {storetype}')
+
+    result = urllib.parse.urlsplit(dst)
+    # the exact mounting point being either the bucket or subdirectory in it
+    sync_point = result.path.split('/')[-1]
+    log_file_name = f'csync_{storetype}_{sync_point}.log'
+    base_dir = os.path.expanduser(_CSYNC_BASE_PATH)
+    os.makedirs(base_dir, exist_ok=True)
+    log_path = os.path.expanduser(os.path.join(base_dir, log_file_name))
```

Comment:
Instead of handling output redirection in this function, can we directly redirect the output in the place we call `python -m sky.data.skystorage csync` by adding `>> ~/.sky/storage.log` , for simplicity

Reply:
Each log file for each `CSYNC` process will be quite large and it will continue to get larger whenever `sync` is ran. Due to that reason, I thought it's necessary to keep the log files separate for each `CSYNC` process ran. And in that case, I'd still need to do the handling to get a different `log_file_name` for each log file while moving this to where we call `python -m sky.data.skystorage csync` with `>> ~/.sky/{log_file_name}`. Should I do this, or just keep all the  log files together as you mentioned above into `~/.sky/storage.log` for simplicity?

Reply:
Great. Made the change in `sky/data/mounting_utils.py/get_mounting_command`